pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -224,6 +225,10 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 				return nil, fmt.Errorf("running with swap on is not supported, please disable swap! or set --fail-swap-on flag to false. /proc/swaps contained: %v", swapLines)
 			}
 		}
+
+		if !util.IsTmpfsNoswapOptionSupported(mountUtil) {
+			klog.InfoS("tmpfs noswap option is not supported, hence memory-backed volumes (e.g. secrets, emptyDirs) might be swapped to disk")
```

Comment:
> I think this will warn on machines where swap is not even enabled

I think this behavior is desired.

If somebody uses failSwapOn + disabled NodeSwap then secrets can still be swapped to disk if a swap file/partition exists on the node.

This bug/security risk existed from whenever the failSwapOn was introduced, before NodeSwap even existed. IOW, this problem is out-of-scope for the NodeSwap feature, but it's rather a `--fail-swap-on` bug.

Reply:
I'm not talking about the NodeSwap feature gate, I'm talking about swap in the OS… if that's not enabled, tmpfs volumes can't swap out, right?

Reply:
Aha, got you.
By "swap enabled in the OS" do you mean that there's an existing swap file/partition? This makes total sense to me :+1:

Reply:
yeah, whatever we're using above to detect swap is enabled so the kubelet can fail if `failSwapOn` is true

Reply:
Thanks for your reviews and clarifications! sounds great to me.
Let me rework the code and let you know when it's ready to review again :+1:

Reply:
@liggitt the code is refactored now and is much clearer IMO.
I believe I've addressed your feedback here.

PTAL